donkeycar/parts/tfmini.py

```diff
@@ -0,0 +1,70 @@
+import time
+import serial
+
+class TFMini:
+    """
+    Class for TFMini and TFMini-Plus distance sensors.
+    See wiring and installation instructions at https://github.com/TFmini/TFmini-RaspberryPi
+
+    Returns distance in centimeters. 
+    """
+
+    def __init__(self, port="/dev/serial0", baudrate=115200, poll_delay=0.01, init_delay=0.1):
+        self.ser = serial.Serial(port, baudrate)
+        self.poll_delay = poll_delay
+
+        self.dist = 0
+        self.strength = 0
+
+        if self.ser.is_open == False:
+            self.ser.close() # in case it is still open, we do not want to open it twice
+            self.ser.open()
+
+        print("Init TFMini")
+        time.sleep(init_delay)
+
+    def update(self):
+        while self.ser.is_open:
+            self.poll()
+            if self.poll_delay > 0:
+                time.sleep(self.poll_delay)
```

Comment:
Why do we want a poll delay here? At a car speed of 4m/s this automatically creates as systematic error of 4cm, using your default value of 10ms in the constructor. I haven't seen any problems with a threaded part running the thread continuously. Did you see any problems when running w/ 0 poll delay? If not, we could safely remove that delay.

Reply:
Yes, without poll delay the part would be able to output only a few values before failing. I'm not entirely sure why this is but I suspect it's because the serial port's input buffer is not clearing fast enough to run without a poll delay. The `self.ser.reset_input_buffer()` helps but doesn't completely solve it.

Reply:
@mgagvani - thanks for explaining, this is very valuable. Can you add the above explanation as a comment into the code, so we don't forget why this is important?